pkg/kubehunter/scanner.go

```diff
@@ -109,6 +112,11 @@ func (s *Scanner) prepareKubeHunterJob() (*batchv1.Job, error) {
 		kubeHunterArgs = append(kubeHunterArgs, "--quick")
 	}
 
+	scanJobAnnotations, err := s.config.(starboard.ConfigData).GetScanJobAnnotations()
```

Comment:
We should inject ConfigData and avoid casting to wider interface

Reply:
Daniel, I guess to tackle this what we can do is that from the struct here, we can remove the `config Config` and add `starboardConfig starboard.ConfigData` in place of it instead.

So, this wouldn't require refactoring any of the existing functionality as `Config` type is an interface and `starboard.ConfigData` struct has the relevant methods to make it implement that interface. Meaning that previously whatever methods were being called upon `config Config` (in the above scanner struct of kubehunter) can be, without any change, called upon by `starboardConfig starboard.ConfigData`. In this case, `GetKubeHunterImageRef()` is the method.

PS: I am suggesting this instead of straight away adding `starboardConfig starboard.configData` without getting rid of `config Config`, because doing so would add unnecessary redundancy as `config Config` wouldn't be serving any special purpose proving its necessity/need in the struct as whatever it is doing can be simply done by `starboardConfig starboard.ConfigData` itself.

I hope that makes sense.

Reply:
Okey. Let's leave it like it is for now. We can refactor it later on.